ipykernel/tests/test_kernelspec.py

```diff
@@ -88,6 +88,25 @@ def test_write_kernel_spec_path():
     shutil.rmtree(path)
 
 
+def test_write_kernel_spec_permissions():
+    read_only_resources = tempfile.mkdtemp()
+    shutil.copytree(RESOURCES, read_only_resources, dirs_exist_ok=True)
```

Comment:
The `dirs_exist_ok` parameter is new in Python 3.8, and ipykernel will need to support earlier versions for a while. The easy way to do that is just to copy to a subdirectory of your new temp directory.

Reply:
Implemented your suggestion